### PR TITLE
Created UI page for application

### LINKED ISSUE #57: Create a single page UI for application
**As a** developer  
**I need** a single page UI for our application
**So that** I can test the wishlist function with front end  

**Assumptions:** 
* use html for front end and javascript for back end

**Acceptance Criteria:**
```
Given I have the UI page setup
When I access the URL for wishlists
Then I could see the UI page successfully
```

### PR DESCRIPTION
**Pull request by xiafanzeng**
image
image

### CODE REVIEW

File: service/static/index.html [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+              <div class="form-group">
+                <label class="control-label col-sm-2" for="wishlist_items">Items:</label>
+                <div class="col-sm-10">
+                  <input type="text" class="form-control" id="wishlist_items" placeholder="Enter items for Wishlist in format [item1, item2, ...]">
+                </div>
+              </div>
+              <div class="form-group">
+                <label class="control-label col-sm-2" for="wishlist_status">Status:</label>
+                <div class="col-sm-10">
+                  <select class="form-control" id="wishlist_status">
```
Comment (abhinavGupta16) [inline service/static/index.html:61]: We can have the status as enabled or disabled.

File: service/static/items.html [hunk 2]
```diff
@@ -0,0 +1,97 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Wishlist RESTful Service</title>
```
Comment (abhinavGupta16) [inline service/static/items.html]: Maybe say `Wishlist RESTful Service (Items)`

File: service/static/index.html [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+        <!-- Search Results -->
+        <div class="table-responsive col-md-10" id="search_results">
+          <table class="table-striped">
+            <thead>
+            <tr>
+                <th class="col-md-2">ID</th>
+                <th class="col-md-2">Name</th>
+                <th class="col-md-2">UserID</th>
+                <th class="col-md-2">Items</th>
```
Comment (abhinavGupta16) [inline service/static/index.html]: I don't think we should have items. How will we enter items in the text field? Maybe wishlist UI can create a wishlist with no items, and then simply use the items UI to run add items?

**Review by abhinavGupta16 — CHANGES_REQUESTED** [review]

**Comment by abhinavGupta16** [comment]
List all wishlists and Items should work without any params. I get an error when I try that. 

image

The problem I think is that you are adding status as a query param, which is wrong.

File: service/static/items.html (hunk 2 above)
Comment (xiafanzeng) [reply service/static/items.html]: > List all wishlists and Items should work without any params. I get an error when I try that.
> 
> image

I just finished UI and wrote some samples in features and restapi.js, but not real interaction. Let's leave the functions to other stories.

Comment (xiafanzeng) [reply service/static/items.html]: > Maybe say `Wishlist RESTful Service (Items)`

Done

File: service/static/index.html (hunk 1 above)
Comment (xiafanzeng) [reply service/static/index.html:61]: Done

**Comment by abhinavGupta16** [comment]
We need to fix the home page test case now I believe.

image

**Comment by xiafanzeng** [comment]
> We need to fix the home page test case now I believe.
> 
> image

Sure I changed the name

File: service/static/index.html (hunk 3 above)
Comment (xiafanzeng) [reply service/static/index.html]: How do you wanna display items in the wishlist page?

Comment (abhinavGupta16) [reply service/static/index.html]: You don't. We can see the wishlists on the wishlist page.... We can go to the items page to see the items I guess.

File: service/static/index.html [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                  <select class="form-control" id="wishlist_status">
+                      <option value="true" selected>Enabled</option>
+                      <option value="false">Disabled</option>
+                  </select>
+                </div>
+              </div>
+              <div class="form-group">
+                <div class="col-sm-offset-2 col-sm-10">
+                  <button type="submit" class="btn btn-primary" id="search-btn">Search</button>
+                  <button type="submit" class="btn btn-primary" id="clear-btn">Clear</button>
```
Comment (abhinavGupta16) [inline service/static/index.html:70]: Clear button does not work

**Review by abhinavGupta16 — CHANGES_REQUESTED** [review]

**Comment by abhinavGupta16** [comment]
> List all wishlists and Items should work without any params. I get an error when I try that.
> 
> image
> 
> The problem I think is that you are adding status as a query param, which is wrong.

@xiafanzeng - I have fixed it and added search functionality.

image

**Review by abhinavGupta16 — COMMENTED** [review]
Should we remove this method altogether? 
`def index():`

**Comment by xiafanzeng** [comment]
> Should we remove this method altogether?
> `def index():`

I once had a mistake loading the homepage so I added this method. I think we can leave it here.

**Comment by abhinavGupta16** [comment]
> > Should we remove this method altogether?
> > `def index():`
> 
> I once had a mistake loading the homepage so I added this method. I think we can leave it here.

Ohh, what mistake? I removed the method because we should not have 2 methods responding to the same url.

File: service/service.py [hunk 5]
```diff
@@ -38,6 +38,16 @@
 from . import app
 
 
+@app.route('/items.html')
+def items():
```
Comment (abhinavGupta16) [inline service/service.py:42]: Can you add docstrings? Also, a test case similar to the '/'

File: service/service.py [hunk 6]
```diff
@@ -38,6 +38,16 @@
...
+    return app.send_static_file('items.html')
+
+
+@app.route('/')
+def homepage():
```
Comment (abhinavGupta16) [inline service/service.py:48]: Can you add docstrings?

**Review by abhinavGupta16 — DISMISSED** [review]

**Review by yf1357 — CHANGES_REQUESTED** [review]
It doesn't work for me.
image
image

**Comment by abhinavGupta16** [comment]
> It doesn't work for me.
> image
> image

Is this nosetests that failed? Try vagrant provision. Nosetests seem to run fine on travis-ci

**Comment by yf1357** [comment]
> > It doesn't work for me.
> > image
> > image
> 
> Is this nosetests that failed? Try vagrant provision. Nosetests seem to run fine on travis-ci

I used provision before and both nosetests,honcho fail. 
Just download and vagrant up this branch again and get the same error.

**Comment by abhinavGupta16** [comment]
> > > It doesn't work for me.
> > > image
> > > image
> > 
> > 
> > Is this nosetests that failed? Try vagrant provision. Nosetests seem to run fine on travis-ci
> 
> I used provision before and both nosetests,honcho fail.
> Just download and vagrant up this branch again and get the same error.

@yf1357 - That is weird. I don't think there is an error with the code.... since the build is successful with test cases in travis.

Maybe vagrant destroy and up ?

**Comment by yf1357** [comment]
> > > > It doesn't work for me.
> > > > image
> > > > image
> > > 
> > > 
> > > Is this nosetests that failed? Try vagrant provision. Nosetests seem to run fine on travis-ci
> > 
> > 
> > I used provision before and both nosetests,honcho fail.
> > Just download and vagrant up this branch again and get the same error.
> 
> @yf1357 - That is weird. I don't think there is an error with the code.... since the build is successful with test cases in travis.
> 
> Maybe vagrant destroy and up ?

Actually I destroyed it lol, I may still have some other setting errors. Just ask Qizhuo to review first

**Comment by sqz0914** [comment]
> > > > It doesn't work for me.
> > > > image
> > > > image
> > > 
> > > 
> > > Is this nosetests that failed? Try vagrant provision. Nosetests seem to run fine on travis-ci
> > 
> > 
> > I used provision before and both nosetests,honcho fail.
> > Just download and vagrant up this branch again and get the same error.
> 
> @yf1357 - That is weird. I don't think there is an error with the code.... since the build is successful with test cases in travis.
> 
> Maybe vagrant destroy and up ?

I got the same error as Yuyang did. 
WechatIMG4

**Comment by abhinavGupta16** [comment]
@yf1357 @sqz0914 - This is so weird. I just tried to destroy and up.

fails on the master branch too. Can you confirm?

**Comment by sqz0914** [comment]
> @yf1357 @sqz0914 - This is so weird. I just tried to destroy and up.
> 
> fails on the master branch too. Can you confirm?

Yeah. I got the same error on the master branch.
Screen Shot 2020-11-12 at 12 19 14 AM

**Comment by abhinavGupta16** [comment]
@yf1357 @sqz0914 - 
There was a problem with vagrant file. I fixed it. Can you try now? 
Irrespective, I am not sure how it works for travis-ci

**Comment by sqz0914** [comment]
> @yf1357 @sqz0914 -
> There was a problem with vagrant file. I fixed it. Can you try now?
> Irrespective, I am not sure how it works for travis-ci

Nice Catch! It works fine now on my computer, as well as `honcho start`. Maybe Travis-CI doesn't use Vagrant. lol.

**Review by yf1357 — APPROVED** [review]
UI page works for me now. I will approve it first.
But there are still something to change for "search"  like the .steps file(cannot pass behave) and the postion of item table(could move to item page). We can change it in search story later

**Review by sqz0914 — APPROVED** [review]

**Comment by abhinavGupta16** [comment]
> UI page works for me now. I will approve it first.
> But there are still something to change for "search" like the .steps file(cannot pass behave) and the postion of item table(could move to item page). We can change it in search story later

Yes, this was only UI. We can complete the rest in other stories.

**Comment by abhinavGupta16** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/b7dd79b7-17cc-412e-98f1-e1a234aafd1c?env_id=ibm:yp:us-south "nyu-wishlist-service-f20-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/76e35b85-cc07-4faf-b3ae-ee3dbe88d89f "nyu-wishlist-service-f20-pipeline") deployed [wishlists](https://github.com/wishlists/wishlists/releases/tag/deploy-ibm-yp-us-south-ag7387-nyu.edu-dev-20201112-192502 "deploy-ibm-yp-us-south-ag7387-nyu.edu-dev-20201112-192502") to [dev](https://console.bluemix.net/apps/f964e7e4-0a9e-4993-92a0-2e294cb63ccd?env_id=ibm:yp:us-south "ibm:yp:us-south:ag7387@nyu.edu:dev"), including commit c0d10147a23150bf444b51e129b62d9d605482ef

**Comment by abhinavGupta16** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/b7dd79b7-17cc-412e-98f1-e1a234aafd1c?env_id=ibm:yp:us-south "nyu-wishlist-service-f20-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/76e35b85-cc07-4faf-b3ae-ee3dbe88d89f "nyu-wishlist-service-f20-pipeline") deployed [wishlists](https://github.com/wishlists/wishlists/releases/tag/deploy-ibm-yp-us-south-ag7387-nyu.edu-prod-20201113-214248 "deploy-ibm-yp-us-south-ag7387-nyu.edu-prod-20201113-214248") to [prod](https://console.bluemix.net/apps/b9e11a39-3222-4ca4-a89b-8f6a16a809d0?env_id=ibm:yp:us-south "ibm:yp:us-south:ag7387@nyu.edu:prod"), including commit c0d10147a23150bf444b51e129b62d9d605482ef

**Pull request merged**